tests/all/func.rs

```
@@ -455,9 +455,7 @@ fn func_write_nothing() -> anyhow::Result<()> {
 }
 
 #[test]
-// Note: Cranelift only supports refrerence types (used in the wasm in this
-// test) on x64.
-#[cfg(target_arch = "x86_64")]
+#[cfg(any(target_arch = "aarch64", target_arch = "x86_64"))]
```

I think this can probably remove the `#[cfg]` now? (it's probably best to keep a "know failing list" rather than a known passing list)

---

Since the CI tests are only on AArch64 and x86-64, that's what I did initially, but then I decided to make it easier for other platforms (and with QEMU it should be straightforward in principle to add s390x tests, for example).

I am happy to go for either approach (and what you say about a known failing list makes sense), but I think I will need to keep some of these tests disabled because right now they seem to fail due to memory exhaustion (everything passes when I run it natively on my AArch64 development machine).

---

Personally I prefer to get failures rather than silent passing on new architectures, and most of the test suite will already fail if you try to test wasmtime on a mips machine for example. Now that being said if these get switched to an ignore for s390x that seems reasonable! 

(it's relatively easy to grep for `TODO.*s390x` or something similar whereas it's a bit more unclear to search for something that only runs for other platforms)

---

`grep -R cfg.*ignore` was my approach :).

---

Heh true! Well either way doesn't matter too much to me, so I'm happy to do whatever you feel is best

---

According to the comment, the reason these were disabled are reference types, but we do in fact fully support reference types on s390x, so those should probably just work ...

---

... and I just verified the tests do indeed pass on s390x when enabled.

---

@uweigand What about the general question - should the tests be ignored or left to fail? E.g. if reference types weren't supported by the s390x backend, from your perspective would it have been preferable to keep a directive here to ignore the tests or not?

---

I think ignore by default is never a good idea.  They should fail by default, and then can be set to ignore on a one-by-one basis per test/platform after analysis.  (Then you can at least grep for your platform name to find a list of tests you still need to work on.)